impl/src/expand.rs

```diff
@@ -235,14 +243,32 @@ fn impl_enum(input: Enum) -> TokenStream {
                     }
                 }
                 (Some(backtrace_field), _) => {
+                    let source = variant.from_field().map(|f| &f.member);
```

Comment:
I don't think `from_field` is the right field to check here &mdash; it should be `source_field`. Otherwise this will generate noncompilable code in the case of `#[source] #[backtrace]` inside an enum, because the generated code would treat the source field as if its type were `Backtrace`.

```suggestion
                    let source = variant.source_field().map(|f| &f.member);
```